File: client-app/src/app/model/OrderAggregate/Order.ts
import { Company } from '../CompanyAggregate/Company';
import { OrderStatus } from './OrderStatus';
import { Cancellation } from './Cancellation';
import { AcceptanceForm } from './AcceptanceForm';
import { Invoice } from './Invoice';
import { ServiceCategory } from '../ServiceCategory';
import { OrderAddress } from './OrderAddress';

export interface Order {
    id: string;
    buyerId: string;
    buyerName: string;
    buyerEmail: string;
    buyerPhone: string;
    buyerMessage: string;
    orderDate: Date;
    orderStatus: OrderStatus;
    serviceCategory: ServiceCategory;
    orderAddresses: OrderAddress[];
    startTime: Date;
    endTime: Date;
    acceptanceForm: AcceptanceForm;
    serviceSchedule: string;
    note: string;
    invoices: Invoice[];
    paymentIntentId: string;
    commissionPercentage: number;
    commission: number;
    cancellation: Cancellation;
    companyId?: string;
    company: Company;
}

export class Order implements Order {
    constructor(init?: OrderFormValues) {
        Object.assign(this, init);
    }
}

export class OrderFormValues {
    id?: string;
    buyerId: string = "";
    buyerName: string = "";
    buyerEmail: string = "";
    buyerPhone: string = "";
    buyerMessage: string = "";
    orderDate: Date = new Date();
    orderStatus: OrderStatus = OrderStatus.Processing;
    serviceCategory: ServiceCategory = ServiceCategory.Moving;
    orderAddresses?: OrderAddress[];
    startTime: Date = new Date();
    endTime: Date = new Date();
    serviceSchedule?: string;
    note?: string;
    companyId?: string;

    constructor(order?: OrderFormValues) {
        if (order) {
            this.id = order.id;
            this.buyerId = order.buyerId;
            this.orderDate = order.orderDate;
            this.orderStatus = order.orderStatus;
            this.serviceCategory = order.serviceCategory;
            this.orderAddresses = order.orderAddresses;
            this.startTime = order.startTime;
            this.endTime = order.endTime;
            this.serviceSchedule = order.serviceSchedule;
            this.note = order.note;
            this.companyId = order.companyId;
        }
    }
}

File: Domain/OrderAggregate/Order.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Domain.CompanyAggregate;
using Domain.InvoiceAggregate;

namespace Domain.OrderAggregate
{
    public class Order
    {
        public Guid Id { get; set; }
        public string BuyerId { get; set; }
        public string BuyerName { get; set; }
        public string BuyerEmail { get; set; }
        public string BuyerPhone { get; set; }
        public string BuyerMessage { get; set; }
        public DateTime OrderDate { get; set; }
        public OrderStatus OrderStatus { get; set; }
        public ServiceCategory ServiceCategory { get; set; }
        public ICollection<OrderAddress> OrderAddresses { get; set; }
        public DateTime StartTime { get; set; }
        public DateTime EndTime { get; set; }
        public AcceptanceForm AcceptanceForm { get; set; }
        public string ServiceSchedule { get; set; }
        public string Note { get; set; }
        public ICollection<Invoice> Invoices { get; set; }
        public string PaymentIntentId { get; set; }
        public long CommissionPercentage { get; set; }
        public long Commission { get; set; }
        public Cancellation Cancellation { get; set; }
        public Guid CompanyId { get; set; }
        public Company Company { get; set; }
    }
}